LibraryInstrumentationGuidance.md

```diff
@@ -0,0 +1,107 @@
+# Guidence for instrumenting libraries with Diagnostic Source 
+
+This document provides guidence for adding Diagnostic Source instrumentation to external libraries, which will allow Application Insights and other monitoring solutions to collect meaningful and rich telemetry.
+
+## Diagnostic Source and Activities
+
+[Diagnostic Source][DiagnosticSourceGuide] is a simple module that allows code to be instrumented for production-time logging of rich data payloads for consumption within the process that was instrumented. At runtime consumers can dynamically discover data sources and subscribe to the ones of interest.
+
+[Activity][ActivityGuide] is a class that allows storing and accessing diagnostics context and consuming it with logging system.
+
+Both Diagnostic Source and Activity have been used to instrument [System.Net.Http][SystemNetHttp] and [Microsoft.AspNetCore.Hosting][MicrosoftAspNetCoreHosting], although that instrumentation is not fully complaiant with this guidance.
+
+More recently two new libraries were instrumented and that work was the basis for this guidance. These libraries are client SDKs for [Azure Event Hubs][MicrosoftEventHubs] and [Azure Service Bus][MicrosoftServiceBus], both of which support very high throuput scenarios.
+
+[This document][DiagnosticSourceActivityHowto] goes into more details on how to efficiently use Diagnostic Source.
+
+## What should be instrumented
+
+The goal of instrumentation is to give users the visibility to how particular operations are being performed inside the library. This information can be later used to diagnose performance or issues. It is up to the library authors to identify operations that the library performs and are worthy of monitoring. Some good examples are external dependency calls, cache utilization (hit/miss/refresh) or awaiting for system events. 
```

Comment:
> Some good examples are external dependency calls, cache utilization (hit/miss/refresh) or awaiting for system events.

This sentence (being correct) gives the wrong impression as this doc targets lib owners. For the libs, we expect some public API operations that cause one or more external dependency calls to be instrumented. I think it should be more clear that it's not a single isolated external dependency call, but the full operation that may consist of several retries, input and output processing.

Reply:
I've changed this description although I am not sure if it conveys exactly what you mean. Please take a look